src/goto-programs/json_goto_trace.cpp

```
@@ -114,6 +111,13 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
+          if(step.full_lhs_value.is_not_nil())
+            full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
```

When does the nil case happen? This looks like trying to mask errors.